### PR TITLE
Fix data cube calculation

### PR DESCRIPTION
**Pull request by jiwalker-usgs**

### CODE REVIEW

**Comment by jiwalker-usgs** [comment]
@isuftin would you mind looking at this too?

File: gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtil.java [hunk 1]
```diff
@@ -195,7 +195,7 @@ public DataCube(int xLength, int yLength, int tLength, int dataTypeSize,
             this.yLength = yLength;
             this.tLength = tLength;
             this.dataTypeSize = dataTypeSize;
-            this.totalSize = xLength * yLength * tLength * dataTypeSize;
+            this.totalSize = (long)xLength * (long)yLength * (long)tLength * (long)dataTypeSize;
```
Comment (isuftin) [inline gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtil.java]: Why do you do long conversion here if a long is of higher precision than an int and you're assigning to a long?

File: gdp-process-wps/src/test/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+	@Test
+	public void testDataCube_size() throws InvalidRangeException, TransformException, FactoryException {
+		// first is under 2 billion, works normally
+		GDPAlgorithmUtil.DataCube dataCubeSmall = new GDPAlgorithmUtil.DataCube(2, 3, 5, 7, null, null, null);
+		// next is over 2 billion, with just 2 dimensions
+		GDPAlgorithmUtil.DataCube dataCubeLarge = new GDPAlgorithmUtil.DataCube(1000000000, 3, 1, 1, null, null, null);
+		// next is over 2 billion only when all dimensions combined
+		GDPAlgorithmUtil.DataCube dataCubeHuge = new GDPAlgorithmUtil.DataCube(1000, 1000, 1000, 4, null, null, null);
+		assertThat(dataCubeSmall.totalSize, is(equalTo(210l)));
```
Comment (isuftin) [inline gdp-process-wps/src/test/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtilTest.java]: OCD. Why not use getters/setters and encapsulate the internal data for this bean?

File: gdp-process-wps/src/test/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * @author jiwalker
+ */
+public class GDPAlgorithmUtilTest {
+	
+	public GDPAlgorithmUtilTest() {
+	}
+
+	@Test
+	public void testDataCube_size() throws InvalidRangeException, TransformException, FactoryException {
```
Comment (isuftin) [inline gdp-process-wps/src/test/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtilTest.java]: OCD. testDataCubeSize

File: gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtil.java (hunk 1 above)
Comment (jiwalker-usgs) [inline gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtil.java]: The intermediate calculations were being stored as int values, and they were overflowing the int range.  The individual values should always be smaller than an int, but when multiplied together may get bigger.

File: gdp-process-wps/src/test/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtilTest.java (hunk 2 above)
Comment (jiwalker-usgs) [inline gdp-process-wps/src/test/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtilTest.java]: Bean seemed like overkill here, really just want some associated values.

File: gdp-process-wps/src/test/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtilTest.java (hunk 3 above)
Comment (jiwalker-usgs) [inline gdp-process-wps/src/test/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtilTest.java]: yeah, I felt a bit weird about this one too.

File: gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtil.java (hunk 1 above)
Comment (isuftin) [inline gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtil.java]: @jiwalker-usgs Makes sense. I didn't think of the calculation done as a pipe in the JVM.

File: gdp-process-wps/src/test/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtilTest.java (hunk 2 above)
Comment (isuftin) [inline gdp-process-wps/src/test/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtilTest.java]: @jiwalker-usgs Why not just have a bean as your default data structure? If you just wanted associated values, having a test for something that simple seems iffy.

Comment (jiwalker-usgs) [inline gdp-process-wps/src/test/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtilTest.java]: This test is really about isolating the int overflow issue I was seeing.  The first assertion is the normal case (full calculation fits within int), the other two are where it was overflowing the int in some way.

File: gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtil.java (hunk 1 above)
Comment (jiwalker-usgs) [inline gdp-process-wps/src/main/java/gov/usgs/cida/gdp/wps/algorithm/GDPAlgorithmUtil.java]: Yeah, it is right to left, so if two (int) values are multiplied they are stored temporarily as (int)s until eventually coercing it to long on assignment.

**Comment by isuftin** [comment]
@jiwalker-usgs  I think you're good to go on the functional aspect of this commit. Up to you regarding naming and encapsulation

**Pull request merged**